@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -79,55 +79,216 @@ export interface VisibleTileSetOptions {
 const MB_FACTOR = 1.0 / (1024.0 * 1024.0);
 
 /**
- * Missing Typedoc
+ * Wrapper for LRU cache that encapsulates tiles caching for any [[DataSource]] used.
+ *
+ * Provides LRU based caching mechanism where each tile is identified by its tile key
+ * (morton code) and data source name.
+ * Tiles are kept in the cache based on last recently used policy, cached tile may be evicted
+ * only when cache reaches full saturation and tile is no longer visible.
+ * @note Currently cached entries (tiles) are identified by unique tile code (morton code) and
+ * data source name, thus it is required that each [[DataSource]] used should have unique
+ * name, but implementation could be improved to omit this limitation.
  */
 class DataSourceCache {
-    readonly tileCache: LRUCache<number, Tile>;
-    readonly disposedTiles: Tile[] = [];
+    static getKey(tileCode: number, dataSource: DataSource): string {
+        return `${dataSource.name}_${tileCode}`;
```

Comment:
as mentioned below, I would prefer following cache key:
`${dataSource.name}_${mortonCode}_${tileOffset}`

Reply:
Because this may be hard change and it inevitably conflicts with renderList hold in VisibleTileSet.dataSourceTileList, I would strongly insist to split this to second PR, cause this change may cause some regression bugs, it will be easier to investigate and revert (if needed).

Reply:
Done in separate patchset: 
https://github.com/heremaps/harp.gl/pull/884